code/modules/reagents/reagent_containers/cups/_cup.dm

```diff
@@ -25,34 +25,33 @@
 		var/list/types = bitfield_to_list(drink_type, FOOD_FLAGS)
 		. += span_notice("It is [lowertext(english_list(types))].")
 
-/obj/item/reagent_containers/cup/proc/checkLiked(fraction, mob/M)
-	if(last_check_time + 50 >= world.time)
-		return
-	if(!ishuman(M))
-		return
-	var/mob/living/carbon/human/H = M
-	if(HAS_TRAIT(H, TRAIT_AGEUSIA))
-		if(drink_type & H.dna.species.toxic_food)
-			to_chat(H, span_warning("You don't feel so good..."))
-			H.adjust_disgust(25 + 30 * fraction)
-	else
-		if(drink_type & H.dna.species.toxic_food)
-			to_chat(H,span_warning("What the hell was that thing?!"))
-			H.adjust_disgust(25 + 30 * fraction)
-			H.add_mood_event("toxic_food", /datum/mood_event/disgusting_food)
-		else if(drink_type & H.dna.species.disliked_food)
-			to_chat(H,span_notice("That didn't taste very good..."))
-			H.adjust_disgust(11 + 15 * fraction)
-			H.add_mood_event("gross_food", /datum/mood_event/gross_food)
-		else if(drink_type & H.dna.species.liked_food)
-			to_chat(H,span_notice("I love this taste!"))
-			H.adjust_disgust(-5 + -2.5 * fraction)
-			H.add_mood_event("fav_food", /datum/mood_event/favorite_food)
-
+// all this copypaste from the edible component is quite deplorable, why the fuck does this not use the edible component?
```

Comment:
Please just replace with a normal autodoc comment (you can still recommend putting it on edible component)